templates/instance.cue

```diff
@@ -32,6 +34,9 @@ import (
 			objects: caInjectorPodDisruptionBudget: #PodDisruptionBudget & {
 				#config:    config
 				#component: "caInjector"
+				if #config.caInjector.podDisruptionBudget.minAvailable != _|_ {
+					#config: caInjector: podDisruptionBudget: minAvailable: <=#config.caInjector.replicas | #Percent
+				}
```

Comment:
Should this be defined in the config or object itself rather than in the instances?

```
        spec: {
                selector: matchLabels: #meta.#LabelSelector
                if #config[#component].podDisruptionBudget.enabled {
                        if #config[#component].podDisruptionBudget.minAvailable != _|_ {
                                minAvailable: #config[#component].podDisruptionBudget.minAvailable & >= #config[#component].replicas
                        }
                        if #config[#component].podDisruptionBudget.maxUnavailable != _|_ {
                                maxUnavailable: #config[#component].podDisruptionBudget.maxUnavailable
                        }
                }
        }
```

Though doesn't seem good to have it required `>=` replicas if someone decides to set replicas to 5 or 10 or whatever.

Reply:
Looking at this again I would actually put the requirements into `#config[#component].podDisruptionBudget` as:

```cue
if config.highAvailability.enabled {
  config: controller: podDisruptionBudget: minAvailable: *controller.replicas | cfg.#Percent | uint16 & >= 1
}

if config.highAvailability.enabled {
  config: controller: podDisruptionBudget: minAvailable: *controller.replicas | cfg.#Percent | uint16 & >= 2
}
```

Reply:
Hmm but this the requires the field to not be optional and could conflict if the user wanted to use `maxUnavailable` instead :thinking:

Reply:
I move check to podDisruptionBudget.cue.
The test check that number of replicas is >= than minAvailable, not the inverse.